src/components/carousel/index.jsx

```diff
@@ -0,0 +1,162 @@
+import { Children, cloneElement, useState } from "react";
+import PropTypes from "prop-types";
+import { useSwipeable } from "react-swipeable";
+
+import Button from "./_children/Button";
+import Item from "./_children/Item";
+
+const COMPONENT_CLASS_NAME = "c-carousel";
+
+const DefaultNextButton = ({ id, onClick }) => (
+	<Button
+		id={id}
+		onClick={onClick}
+		label="Next Slide"
+		className="c-carousel__button c-carousel__button--next"
+	>
+		Next
+	</Button>
+);
+
+const DefaultPreviousButton = ({ id, onClick }) => (
+	<Button
+		id={id}
+		onClick={onClick}
+		label="Previous Slide"
+		className="c-carousel__button c-carousel__button--previous"
+	>
+		Previous
+	</Button>
+);
+
+const Carousel = ({
+	children,
+	className,
+	id,
+	label,
+	next,
+	nextButton,
+	previousButton,
+	previous,
+	slidesToShow,
+	...rest
+}) => {
+	const [slide, setSlide] = useState(slidesToShow);
+	const [position, setPosition] = useState(0);
+	const containerClassNames = [COMPONENT_CLASS_NAME, className].filter((i) => i).join(" ");
+
+	const subComponents = Object.values(Carousel).map((subcomponentType) =>
+		Children.map(children, (child) => (child?.type === subcomponentType ? child : null))
+	);
+
+	const childItems = Children.toArray(subComponents);
+
+	const carouselItems = childItems.map((child, index) => {
+		const viewable = index + 1 > slide - slidesToShow && index + 1 <= slide;
+		return child.type === Item ? cloneElement(child, { viewable }) : null;
+	});
+
+	const previousSlide = () => {
+		if (slide - 1 < slidesToShow) {
+			return;
+		}
+		setSlide(slide - 1);
+		setPosition(position + 100 / slidesToShow);
+	};
+
+	const nextSlide = () => {
+		if (slide + 1 > carouselItems.length) {
+			return;
+		}
+		setSlide(slide + 1);
+		setPosition(position - 100 / slidesToShow);
+	};
+
+	/* istanbul ignore next */
+	const handlers = useSwipeable({
+		onSwipedLeft: () => nextSlide(),
+		onSwipedRight: () => previousSlide(),
+		preventDefaultTouchmoveEvent: true,
+		trackMouse: true,
+	});
```

Comment:
Part 1 of touch fix:
```suggestion
	const handlers = useSwipeable({
		onSwipedLeft: () => nextSlide(),
		onSwipedRight: () => previousSlide(),
		preventDefaultTouchmoveEvent: true,
		trackMouse: true,
		trackTouch: true,
	});
```